ksql-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonExtractStringKudf.java

```diff
@@ -26,31 +26,35 @@
 import java.io.IOException;
 import java.util.List;
 
+import io.confluent.ksql.function.UdfUtil;
 import io.confluent.ksql.function.KsqlFunctionException;
 import io.confluent.ksql.function.udf.Kudf;
+import io.confluent.ksql.util.KsqlException;
 import io.confluent.ksql.util.json.JsonPathTokenizer;
 
 public class JsonExtractStringKudf implements Kudf {
   private static final ObjectReader OBJECT_READER = new ObjectMapper().reader();
+  public static final String NAME = "EXTRACTJSONFIELD";
 
   private List<String> tokens = null;
 
   @Override
-  public Object evaluate(Object... args) {
-    if (args.length != 2) {
-      throw new KsqlFunctionException("getStringFromJson udf should have two input arguments:"
-                                      + " JSON document and JSON path.");
-    }
+  public Object evaluate(final Object... args) {
+    UdfUtil.ensureCorrectArgs("EXTRACTJSONFIELD", args, String.class, String.class);
```

Comment:
just a thought, but rather than having to call this in every one of these functions, could we wrap these classes that does this call first and then delegates to the actual impl?

Reply:
I did think along these lines.   It would be nice if the `Kudf` interface was richer, e.g. had methods to return the name of the function, the accepted arg types, the return type, etc.  (A bit more towards `KsqlFunction`).  Currently this metadata is spread across the function registry and the UDFs themselves.

As for moving this checks out of the UDFs... well, we find the UDF based on argument types already, though the metadata held else where can get out of whack with the UDF.  Having the check here is more defensive and testable in a unit test. Long term moving the metadata into the UDF will remove the possibility of it getting out of whack.

Reply:
The counter argument to the more testable thing is that this could be handled in one place and that is the only thing that needs to be tested. I'm happy either way

Reply:
I think we fix this by migrating these UDFs to be strongly typed and using your UDF loading.